### ISSUE TITLE
[Accessibility] NavigationDestination's label does not appear to scale when operating system font is scaled

### ISSUE THREAD

**Issue by neiljaywarner** [flutter/flutter#132335]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Use bottom navigation bar sample flutter create --sample=material.BottomNavigationBar.1 mysample or seee: https://github.com/neiljaywarner/bottomnavbar/blob/master/lib/main.dart
2. Increase test scaling factor with the OS.

iOS example
Settings->Accessibility-Display&Text Size-> Larger Text, use slider on the bottom.
If you use the toggle "Larger Accessibility Sizes" it can go up to 3.0 at the far right.

### Expected results

The "Page 1" text *and* the "Explore", "Commute" and "Saved' text gets larger

### Actual results

Only the "Page 1" text gets larger.

### Code sample

https://github.com/neiljaywarner/bottomnavbar/blob/master/lib/main.dart
<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [NavigationBar].

void main() => runApp(const NavigationBarApp());

class NavigationBarApp extends StatelessWidget {
  const NavigationBarApp({super.key});

  @override
  Widget build(BuildContext context) {
    double tsf = MediaQuery.of(context).textScaleFactor;
    debugPrint('********* tsf= $tsf');
    return MaterialApp(
        theme: Theme.of(context).copyWith(useMaterial3: true),
        home: NavigationExample(
          title: tsf.toString(),
        ));
  }
}

class NavigationExample extends StatefulWidget {
  const NavigationExample({super.key, required this.title});
  final String title;
  @override
  State<NavigationExample> createState() => _NavigationExampleState();
}

class _NavigationExampleState extends State<NavigationExample> {
  int currentPageIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      bottomNavigationBar: NavigationBar(
        onDestinationSelected: (int index) {
          setState(() {
            currentPageIndex = index;
          });
        },
        selectedIndex: currentPageIndex,
        destinations: const <Widget>[
          NavigationDestination(
            icon: Icon(Icons.explore),
            label: 'Explore',
          ),
          NavigationDestination(
            icon: Icon(Icons.commute),
            label: 'Commute',
          ),
          NavigationDestination(
            selectedIcon: Icon(Icons.bookmark),
            icon: Icon(Icons.bookmark_border),
            label: 'Saved',
          ),
        ],
      ),
      body: <Widget>[
        Container(
          color: Colors.red,
          alignment: Alignment.center,
          child: const Text('Page 1'),
        ),
        Container(
          color: Colors.green,
          alignment: Alignment.center,
          child: const Text('Page 2'),
        ),
        Container(
          color: Colors.blue,
          alignment: Alignment.center,
          child: const Text('Page 3'),
        ),
      ][currentPageIndex],
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

<img width="300" src="https://github.com/flutter/flutter/assets/3019617/615c50e7-28e5-4207-b233-bd1f082ce06f" />
<img width="300" src="https://github.com/flutter/flutter/assets/3019617/7b8728e6-e3f1-4aaf-8d79-9171694d02cf" />

</details>

### Logs

<details><summary>Logs</summary>

```console
Performing hot reload...
Syncing files to device iPhone 14 Pro...
flutter: ********* tsf= 0.8823529411764706
Reloaded 1 of 665 libraries in 411ms (compile: 34 ms, reload: 136 ms, reassemble: 197 ms).
flutter: ********* tsf= 0.8823529411764706
flutter: ********* tsf= 0.8823529411764706
flutter: ********* tsf= 0.8823529411764706
flutter: ********* tsf= 0.8823529411764706
flutter: ********* tsf= 1.3529411764705883
flutter: ********* tsf= 1.3529411764705883
flutter: ********* tsf= 1.3529411764705883
flutter: ********* tsf= 3.1176470588235294
flutter: ********* tsf= 3.1176470588235294
flutter: ********* tsf= 3.1176470588235294
flutter: ********* tsf= 1.9411764705882353
flutter: ********* tsf= 1.1176470588235294
flutter: ********* tsf= 1.1176470588235294
flutter: ********* tsf= 1.1176470588235294
flutter: ********* tsf= 1.0
flutter: ********* tsf= 1.0
flutter: ********* tsf= 1.0
flutter: ********* tsf= 1.0
flutter: ********* tsf= 1.0
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
neil.warner@NA-081MD6TMBP dev % flutter doctor -v
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.5 22G74 darwin-x64, locale en-US)
    • Flutter version 3.10.6 on channel stable at /Users/neil.warner/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (4 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/neil.warner/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • Java binary at: /usr/local/opt/openjdk@11/bin/java
    • Java version OpenJDK Runtime Environment Homebrew (build 11.0.15+0)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.79.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (4 available)
    • SM G960U1 (mobile)     • 594b4d3841563398                     • android-arm64  • Android 10 (API 29)
    • iPhone 14 Pro (mobile) • 44BB58CD-A324-4358-B700-015DC6313498 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 115.0.5790.170

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
neil.warner@NA-081MD6TMBP dev % 
```

</details>

**Comment by danagbemava-nc** [comment]
Hi @neiljaywarner, I don't think it is supposed to scale. I looked at the AppStore bottom navigation after increasing the textSize to the largest (I enabled large text in accessibility as well) and the label looks the same. Am I missing something? Kindly see the screenshots below

| flutter | appstore | 
| -- | -- |
| <img src="https://github.com/flutter/flutter/assets/88313112/71ab0ec7-fd06-44ed-8a81-f117d2368b6c"  /> | <img src="https://github.com/flutter/flutter/assets/88313112/12e05052-1d47-4fd8-9c00-96d9cb08536d" /> |

**Comment by neiljaywarner** [comment]
That's not surprising to me but *why* isn't it supposed to scale is my real
question

The app store reference helps thank you but in reference to wcag a11y 508
and QA perspective...

Ort apple hig or material design..



....sent from my phone

**Comment by danagbemava-nc** [comment]
Hi @neiljaywarner, can you share the resources, I looked at the HIG but I couldn't find what the accessibility behavior for the tab bar should be. On android, the label does scale. I'm also not familiar with wcag a11y 508 so if you could point me as to where to look, I'd appreciate it.

**Comment by neiljaywarner** [comment]
i also do not know what apple HIG guidelines are.
https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html

https://www.a11yproject.com/posts/resize-text/

https://developer.android.com/guide/topics/ui/accessibility
https://docs.flutter.dev/ui/accessibility-and-localization/accessibility#large-fonts
pleaser consider including screenshots showing android scales; that is
interesting; i apologize if i forgot to check that.
https://developer.apple.com/design/human-interface-guidelines/accessibility

**Comment by danagbemava-nc** [comment]
Kindly see the recording below. Playstore scales the text better than the flutter app, but the flutter still scales the label to a certain extent. 

https://github.com/flutter/flutter/assets/88313112/812b2adc-4e04-42f5-9f23-4e64fc0335df

There doesn't seem to be any recommendation on what to do with the labels, however, since I'm not an accessibility expert, I'll leave this for further insight from the team.
